data/wicked/check_macvtap.c

```diff
@@ -45,6 +46,24 @@ char fn[FILENAME_LENGTH];
 // Read buffer
 unsigned char buffer[PACKET_SIZE];
 
+void read_macaddr() {
+#define SYSFS_MAC "/sys/class/net/macvtap1/address"
+  unsigned char *mac = (unsigned char*) &arp_packet[DEST_MAC1_POS_IN_ARRAY];
+
+  FILE *fp = fopen(SYSFS_MAC, "r");
+  if (fp == NULL){
+    perror("Cannot open " SYSFS_MAC);
```

Comment:
mmm , I was not writing C code for ages but I assume that is not how string concatenation works in C ...

Reply:
It is, if you have two strings, you can just concatinate them. I think it is made by the preprocessor, so it is done during compile time, *not* runtime.

Reply:
see here: https://onlinegdb.com/NLe9ls3cI